### ISSUE TITLE
German translation - locales.php

### ISSUE THREAD

**Issue by q2apro** [jacobwb/hashover#6]
'de' => array(
		'comment_form'	=> 'Kommentar hier eingeben (andere Felder sind optional)',	// "Comment" field's default text
		'reply_form'	=> 'Antwort hier eingeben (andere Felder sind optional)',	// "Reply" field's default text
		'post_button'	=> 'Kommentar posten',				// "Post Comment" button's default text
		'del_note'	=> 'Dieser Kommentar wurde gelöscht.',		// Notice of deleted comment
		'cmt_deleted'	=> 'Kommentar gelöscht',				// Notice of successful comment deletion
		'options'	=> 'Optionen',					// "Options" button text
		'cancel'	=> 'Abbrechen',					// "Cancel" button text
		'reply_to_cmt'	=> 'Auf Kommentar antworten',
		'edit_your_cmt'	=> 'Kommentar bearbeiten',
		'nickname_tip'	=> 'Dein Name oder Twitter @name',
		'nickname'	=> 'Name oder @Benutzer',
		'password_tip'	=> 'Passwort (nur falls du den Kommentar später bearbeiten möchtest)',
		'password'	=> 'Passwort',
		'email'		=> 'E-Mail-Adresse',
		'website'	=> 'Webseite',
		'logged_in'	=> 'Du hast dich erfolgreich angemeldet.',
		'cmt_needed'	=> 'Du hast den Kommentar falsch eingegeben. Bitte nutze das Formular unten.',
		'reply_needed'	=> 'Du hast die Antwort falsch eingegeben. Bitte nutze das Formular unten.',
		'post_fail'	=> 'Ups, du hast nicht die erforderlichen Rechte!',
		'cmt_tip'	=> 'Akzeptiertes HTML: &lt;b&gt;, &lt;u&gt;, &lt;i&gt;, &lt;s&gt;, &lt;pre&gt;, &lt;ul&gt;, &lt;ol&gt;, &lt;li&gt;, &lt;blockquote&gt;, &lt;code&gt; wird als HTML dargestellt, URLs werden automatisch zu Links umgewandelt und mit [img]Bild-URL[/img] werden externe Bilder angezeigt.',
		'post_reply'	=> 'darauf antworten',
		'delete'	=> 'Löschen',
		'subscribe_tip'	=> 'E-Mail Benachrichtigung abonnieren',
		'subscribe'	=> 'Abonnieren',
		'edit_cmt'	=> 'Kommentar bearbeiten',
		'save_edit'	=> 'Speichern',
		'no_email_warn'	=> 'Du wirst bei neuen Kommentaren keine Benachrichtigung erhalten, wenn du keine E-Mail angibst.',
		'invalid_email'	=> 'Die von dir eingegebene E-Mail-Adresse ist ungültig.',
		'delete_cmt'	=> 'Möchtest du diesen Kommentar wirklich löschen?',
		'post_cmt'	=> 'Kommentar posten',
		'popular_cmts'	=> 'Beliebteste',
		'showing_cmts'	=> 'Anzeige von',
		'sort'		=> 'Sortieren',
		'sort_ascend'	=> 'In Reihenfolge',
		'sort_descend'	=> 'In umgekehrter Reihenfolge',
		'sort_byname'	=> 'Nach Kommentator',
		'sort_bydate'	=> 'Nach Datum (neueste zuerst)',
		'sort_bylikes'	=> 'Nach Likes',
		'thread'	=> 'Anfang des Threads',
		'thread_tip'	=> 'Spring zum Anfang des Threads',
		'edit'		=> 'Bearbeiten',
		'reply'		=> 'Antworten',
		'like'		=> 'Like',
		'liked'		=> 'Gefällt dir',
		'like_cmt'	=> 'Diese Kommentar \'gefällt mir\'',
		'liked_cmt'	=> 'Dir \'gefällt\' dieser Kommentar',
		'op_cmt_note'	=> 'Du wirst nicht per E-Mail benachrichtigt',
		'subbed_note'	=> 'wirst per E-Mail benachrichtigt',
		'unsubbed_note' => 'hat E-Mail-Benachrichtigungen nicht abonniert'
	),

**Note: **
I have seen that two language strings are hardcoded in javascript-mode.php: 

- **on** in `var pagetitle = (document.title != '') ? ' on "'+ document.title +'"' : '';`
- **Be the first to comment!** in `echo $this->escape_output('<b class="hashover-title">Be the first to comment!</b>\n</div>');`

**Comment by jacobwb** [comment]
Thank you very much for the translation!

As for the hardcoded language strings: it's already fixed in the code I'm hopefully going to push sometime soon, this or next month I hope.

I have a few questions about the translation, though.

1. Is "Antwort" the same as "reply" and "Antworten" the same as "replies"?

  I ask because I see "Antwort" used where "reply" is and "Antworten" used where "replies" is, but I also see "Antworten" used in contexts about actions, for example the "Reply" button is "Antworten".

2. Would "Antworten" be used for "Replies", like would "including replies" be "Zählen Antworten"?

3. Is there a reason "like" is used for the link, but "Gefällt dir" is used for the tooltip?

  In English "Like" is an action all it's own, in this context it doesn't mean the same thing as "I like this" it's more like "thumbs up". Aside from that, I think either "Like" should always be used or "Gefällt" or a better German word should always be used, not both.

4. Is "Nicht Gefällt" an acceptable way to say "dislike" in German, or is there something better?

5. What are the appropriate German words for? ...

  - today,
  - day,
  - days,
  - month,
  - months,
  - year,
  - years,
  - ago

**Comment by q2apro** [comment]
> 1. Is "Antwort" the same as "reply" and "Antworten" the same as "replies"?
Basically in German we use "Antworten" (verb) for "(to) answer" and "(to) reply", we have no word for reply than "antworten". 
For the noun: answers = Antworten, 1 answer = 1 Antwort

> Would "Antworten" be used for "Replies", like would "including replies" be "Zählen Antworten"?
I could not find the language string "including replies" in locales.php. I have to see the context. If you mean, e.g. "15 posts including replies" it gets tricky. It would be "15 Posts inklusive Antworten (auf Kommentare)". 

> Is there a reason "like" is used for the link, but "Gefällt dir" is used for the tooltip?
"Like" has been implemented in the German language, at least in the Internet. However, you can change "Like" to "Gefällt mir" in this case. 

> Is "Nicht Gefällt" an acceptable way to say "dislike" in German, or is there something better?
"Gefällt mir nicht."

In another project I used "Find ich gut" and "Find ich nicht gut". Personally, this is my favorite but most websites orientate what facebook is doing, and they use "Gefällt mir".

> What are the appropriate German words for? ...
today = heute
day = Tag
days = Tage
month = Monat
months = Monate
year = Jahr
years = Jahre
ago = vor

Important:

3 years ago = vor 3 Jahren

we change the order, which could it make tricky for you. 

This is why we use in the q2a project language strings like "^when^where^who" etc. to be changed in order if necessary. See https://github.com/q2a/question2answer/blob/master/qa-include/lang/qa-lang-main.php#L94 and https://github.com/q2a/question2answer/blob/master/qa-include/qa-theme-base.php#L1800

**Comment by jacobwb** [comment]
So "Gefällt mir" for both present and past tense?

When is the ending "n" added, as a plural, when there is more than one?

**Comment by q2apro** [comment]
> So "Gefällt mir" for both present and past tense?
I guess with present you mean: User can click (has not yet), and past: User has clicked (liked). Then it would be present: "Gefällt mir" and past "Gefällt mir nicht mehr" (just checked the facebook labelling for taking the like back).

> When is the ending "n" added, as a plural, when there is more than one?
Yes, plural and preposition "vor" (before). We say "before 3 years" instead of "3 years ago". 
vor 0 Tagen == heute (today)
vor 1 Tag = 1 day ago
vor 2 Tagen = 2 days ago
vor 3 Tagen = 3 days ago
etc.

**Comment by jacobwb** [comment]
Thank you very much for the clarification.

**Comment by q2apro** [comment]
One more idea: You could provide translations like that: 

	EN: 1 year ago
	DE: vor 1 Jahr

	EN: ^ years ago
	DE: vor ^ Jahren

Using the ^ placeholder you get more independent and internationalization gets easier.

**Comment by q2apro** [comment]
By accident I deleted my tip. Guess you have got an email with the text? Please post it here, thanks.

**Comment by jacobwb** [comment]
Yes, that is more or less what is implemented in my current code, except I use `_NUM_` for clarity.

**Comment by jacobwb** [comment]
@q2apro

I just pushed my latest code. I've added some more strings to all the locales, so I had to add some German ones myself, and I think I may have edited a few as well. If you would be so kind as to proofread the German translation for any errors, that would be much appreciated.
